src/Bicep.Core/Emit/InlineDependencyVisitor.cs

```diff
@@ -198,29 +198,44 @@ private void VisitPropertyAccessSyntaxInternal(PropertyAccessSyntax syntax)
                 return;
             }
 
-            static bool ShouldSkipInlining(ObjectType objectType, string propertyName)
+            static bool ShouldSkipInlining(ObjectType objectType, string propertyName, ResourceDeclarationSyntax? resourceSyntax = null)
             {
                 if (!objectType.Properties.TryGetValue(propertyName, out var propertyType))
                 {
                     // unknown property - we should assume it's not inlinable
                     return true;
                 }
 
-                // update the cache if property can't be skipped for inlining
-                return propertyType.Flags.HasFlag(TypePropertyFlags.DeployTimeConstant);
+                if (propertyType.Flags.HasFlag(TypePropertyFlags.DeployTimeConstant))
+                {
+                    if (resourceSyntax is not null &&
+                        propertyName != LanguageConstants.ResourceIdPropertyName &&
+                        propertyName != LanguageConstants.ResourceNamePropertyName &&
+                        propertyName != LanguageConstants.ResourceTypePropertyName &&
+                        propertyName != LanguageConstants.ResourceApiVersionPropertyName &&
```

Comment:
Mind adding a comment here explaining why the special-case check for these properties, with a link to the codegen in `ConvertResourcePropertyAccess`?

Reply:
Yes. Added a comment.